### ISSUE TITLE
Docker Image is a bit outdated

### ISSUE THREAD

**Issue by cmichallek** [syslog-ng/syslog-ng#2941]
Hi,

it would be nice if the Images over at https://hub.docker.com/r/balabit/syslog-ng
would be more recent.

Bonus would be to have the Versions as Tags, so you can switch easily between versions.

greetings

Christian

**Comment by alltilla** [comment]
Hi Christian,

Building the docker images is currently a manually triggered task.
I will make sure to build latest version soon, and automate the process.

Regards,
Attila

**Comment by szemere** [comment]
Hello,
 while @alltilla is working on the automatization, I tagged the current latest image (it was a **3.21.1**), triggered a rebuild and also tagged it, so the current **latest** == **3.24.1**, it is available under: `docker pull balabit/syslog-ng:3.24.1`

Br,
Laci

**Comment by cmichallek** [comment]
thanks alot!
updated my containers and everything works fine.

**Comment by szemere** [comment]
Hello,
 an another manual update happened. Now you can reach:
`balabit/syslog-ng:3.25.1`
`balabit/syslog-ng:3.26.1` (which is also the "latest")

Thank you @nikobearr for reporting it!

Br,
Laci

**Comment by nikobearrr** [comment]
Hi @szemere. 
Would it be possible to push a 3.27.1 docker version?

**Comment by szemere** [comment]
Hi @nikobearrr ,
thank you for asking! You can find it here: https://hub.docker.com/layers/balabit/syslog-ng/3.27.1/images/sha256-2f2d58b646d802146111e5f72dee42b74c8c83eb28d41a772ac77e8c3a4319bf?context=repo

Br,
Laci

**Comment by nikobearrr** [comment]
I guess I was a bit late :)
Thank you @szemere

**Comment by nikobearrr** [comment]
Hi @szemere, is it possible to push the 3.29.1 to dockerhub? Thank you

**Comment by szemere** [comment]
Hello @nikobearrr,
 as you can see, the process is still not automated (hence the issue is still open). Thank you for reminding me! The image is here: https://hub.docker.com/layers/balabit/syslog-ng/3.29.1/images/sha256-73da5ef60c1fe44dec7bdf84221a3d25193a7ae0c948ea19c8f6506d6d8979c1?context=repo

Br,
Laci

**Comment by loganmc10** [comment]
Same request for 3.31.1?

**Comment by szemere** [comment]
Hello,
 sorry for the delay. I am working on the automation, and wanted to create the new version with the new system. Since I can not work on this for the next couple of days, I updated the image manually.

https://hub.docker.com/layers/140762468/balabit/syslog-ng/3.31.1/images/sha256-63f22077faa18961f001c569aa89afdb50a245c6ec605c8739df7aebac529eca?context=explore

Br,
Laci

**Comment by loganmc10** [comment]
Same request for 3.33 :)

**Comment by szemere** [comment]
Hello @loganmc10 ,

this docker file is based on László Budai's unofficial repository, which is discontinued after 3.31.1. (see more [here](https://lists.balabit.hu/pipermail/syslog-ng/2021-June/026292.html) )

From this point there are multiple options to choose from, while we are working on a final solution for this image.

- Download one of the "nightly" artifacts from GitHub. Our CI system constantly testing if we do not break the package building process. As a side product we have binary packages as GitHub Actions Artifacts. i.e.: https://github.com/syslog-ng/syslog-ng/actions/workflows/packages.yml --> https://github.com/syslog-ng/syslog-ng/actions/runs/1020807289

- Build your own package locally: with dbld it is possible to build binary packages from any source. (Including local modifications! If you want to tweak Syslog-ng.) See dbld's README page for further information. https://github.com/syslog-ng/syslog-ng/tree/master/dbld#syslog-ng-developmentrelease-environment-powered-by-docker

- Use an other unofficial repository. (Yes, I can see the irony.) Peter Czanik maintains an unofficial repository of Syslog-ng packages: https://copr.fedorainfracloud.org/coprs/czanik/ , more on that here: https://www.syslog-ng.com/community/b/blog/posts/installing-latest-syslog-ng-on-rhel-and-other-rpm-distributions

Br,
Laci

**Comment by loganmc10** [comment]
Since you're already creating the .deb files, would it be possible to include them in the GitHub release as an asset? That makes them easy to download (you have to be logged into GitHub to download Actions artifacts, so it's hard to automate).

On the same note, couldn't the official Dockerfile (https://github.com/balabit/syslog-ng-docker/blob/master/syslog-ng/Dockerfile) just then use that .deb (from the GitHub release) as it's source?

**Comment by szemere** [comment]
Hello,

I am very grateful for your ideas! It is always good to hear some fresh thoughts. Please, let me share with you (and with others) my progress in the topic:

---
### Add Binary packages to the release artifacts

Currently we are building packages for five platforms, here is a list for Debian Buster:

```
syslog-ng_3.33.1.22.g59d17f5-snapshot+20210715T210325_all.deb
syslog-ng-core_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-dbg_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-dev_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-add-contextual-data_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.de
syslog-ng-mod-amqp_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-examples_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-geoip2_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-getent_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-graphite_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-http_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-map-value-pairs_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-mongodb_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-python_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-redis_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-riemann_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-slog_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-smtp_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-snmp_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-sql_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-stardate_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-stomp_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb
syslog-ng-mod-xml-parser_3.33.1.22.g59d17f5-snapshot+20210715T210325_amd64.deb

```

So overall we have more than one hundred files, per release. I think this will "pollute" the release artifacts, and downloading them one by one...

An alternative solution would be to publish them in the archive / distribution form, like in the CI pipeline. Putting packages into archives is a little bit odd IMHO. And I think it is a sign of the fact, that release artifacts is NOT a package manager.

---

### Using Binary packages in the "official Dockerfile"

(I use quotation marks since that image is not official according to the DockerHub terminology.)

Since there is no alias for the "latest" run in GitHub Actions, I can not see a good way to pinpoint those packages inside the Dockerfile. We need a way to connect Docker image versions with GitHub Actions Job ID's. Or we have to update the URL inside the Dockerfile by hand for each release.

Currently I am working on a change where we would build that image on GitHub with GitHub Actions. (And upload them to Dockerhub for backward compatibility.) This way, we can compile the Binary packages and the Dockerfile in a single job. Eliminating the need for handling Job ID's.

**Comment by MrAnno** [comment]
Fixed by #3870
